File: Selenium/Activities/Activity17.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;

public class Activity17 {
    public static void main(String[] args) {
        WebDriverManager.firefoxdriver().setup();

        // Suppressing browser log
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "NULL");
        // Initialize firefox driver
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));

        // Open a page
        driver.get("https://training-support.net/selenium/selects");

        // Page title
        System.out.println("Title of page is: " + driver.getTitle());

        WebElement dropdown = driver.findElement(By.id("single-select"));
        Select select = new Select(dropdown);

        // Select the second option using the visible text
        select.selectByVisibleText("Option 2");
        System.out.println("Second option: " + select.getFirstSelectedOption().getText());

        //Select the third option using the index
        select.selectByIndex(3);
        System.out.println("Third option: " + select.getFirstSelectedOption().getText());

        //Select the fourth option using the value
        select.selectByValue("4");
        System.out.println("Fourth option: " + select.getFirstSelectedOption().getText());

        // Get all the options
        List<WebElement> options = select.getOptions();
        for (WebElement opt: options) {
            System.out.println(opt.getText());
        }

        driver.quit();
    }
}

File: Selenium/Activities/Activity17.py
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the link in browser
    driver.get("https://training-support.net/selenium/selects")

    # Print page title
    print("Page Title is: " + driver.title)

    dropdown = driver.find_element(By.ID, "single-select")
    dropdown_object = Select(dropdown)

    # Select the second option using the visible text
    dropdown_object.select_by_visible_text("Option 2")
    print("Second option: " + dropdown_object.first_selected_option.text)

    # Select the third option using the index
    dropdown_object.select_by_index(3)
    print("Third option: " + dropdown_object.first_selected_option.text)

    # Select the fourth option using the value
    dropdown_object.select_by_value("4")
    print("Fourth option: " + dropdown_object.first_selected_option.text)

    # Get all the options
    options = dropdown_object.options
    for opt in options:
        print(opt.text)

    driver.quit();